test/uk/gov/hmrc/vatsubscription/controllers/UpdateContactPreferenceControllerSpec.scala

```diff
@@ -0,0 +1,139 @@
+import uk.gov.hmrc.vatsubscription.helpers.UpdateVatSubscriptionTestConstants.{updateErrorResponse, updateSuccessResponse}
+import uk.gov.hmrc.vatsubscription.models.post.CommsPreferencePost
+import uk.gov.hmrc.vatsubscription.models.updateVatSubscription.response.ErrorModel
+import uk.gov.hmrc.vatsubscription.models.{DigitalPreference, PaperPreference}
+import uk.gov.hmrc.vatsubscription.service.mocks.{MockUpdateContactPreferenceService, MockVatCustomerDetailsRetrievalService}
+
+import scala.concurrent.Future
+
+class UpdateContactPreferenceControllerSpec extends TestUtil with MockVatAuthorised
+  with MockUpdateContactPreferenceService with MockVatCustomerDetailsRetrievalService {
+
+  object TestCommsPreference extends UpdateContactPreferenceController(
+    mockVatAuthorised,
+    mockUpdateContactPreferenceService,
+    mockVatCustomerDetailsRetrievalService,
+    controllerComponents)
+
+  val paperPref: FakeRequest[AnyContentAsJson] = FakeRequest().withJsonBody(Json.obj("commsPreference" -> "PAPER"))
+  val digitalPref: FakeRequest[AnyContentAsJson] = FakeRequest().withJsonBody(Json.obj("commsPreference" -> "DIGITAL"))
+
+  "the updateContactPreferences() method" when {
+
+    "the user is not authorised" should {
+
+      "return FORBIDDEN" in {
+
+        mockAuthorise(vatAuthPredicate, retrievals)(Future.failed(InsufficientEnrolments()))
+        val res: Result = await(TestCommsPreference.updateContactPreference(testVatNumber)(paperPref))
+        status(res) shouldBe FORBIDDEN
+      }
+    }
+
+    "the user is authorised" should {
+
+      "return a successful response" when {
+
+        "Paper Preference is supplied and the response from the UpdateVatSubscription service is successful" in {
+
+          mockAuthRetrieveMtdVatEnrolled(vatAuthPredicate)
+          mockExtractWelshIndicator(testVatNumber)(Future(Right(false)))
+          mockUpdateContactPreference(CommsPreferencePost(PaperPreference))(Future.successful(Right(updateSuccessResponse)))
+
+          val res: Result = await(TestCommsPreference.updateContactPreference(testVatNumber)(paperPref))
+
+          status(res) shouldBe OK
+          jsonBodyOf(res) shouldBe Json.toJson(updateSuccessResponse)
+        }
+
+        "Digital Preference is supplied ant the response from the UpdateVatSubscription service is successful" in {
+
+          mockAuthRetrieveMtdVatEnrolled(vatAuthPredicate)
+          mockExtractWelshIndicator(testVatNumber)(Future(Right(false)))
+          mockUpdateContactPreference(CommsPreferencePost(DigitalPreference))(Future.successful(Right(updateSuccessResponse)))
+
+          val res: Result = await(TestCommsPreference.updateContactPreference(testVatNumber)(digitalPref))
+
+          status(res) shouldBe OK
+          jsonBodyOf(res) shouldBe Json.toJson(updateSuccessResponse)
+        }
+      }
+
+      "return an error response" when {
+
+        "no json body is supplied for the PUT" should {
```

Comment:
As mentioned above, need a test for JSON being invalid/comms preference not being digital or paper (once fixed) - manually testing returns part of an exception in the response message